storm-core/src/jvm/backtype/storm/scheduler/TopologyDetails.java

```diff
@@ -433,4 +461,49 @@ private void initConfigs() {
     public Double getTopologyWorkerMaxHeapSize() {
         return this.topologyWorkerMaxHeapSize;
     }
+
+    /**
+     * Get the user that submitted this topology
+     */
+    public String getTopologySubmitter() {
+        String user = (String) this.topologyConf.get(Config.TOPOLOGY_SUBMITTER_USER);
+        if (user == null || user.equals("")) {
+            LOG.debug("Topology {} submitted by anonymous user", this.getName());
+            user = "anonymous";
+        }
+        return user;
+    }
+
+    /**
+     * get teh priority of this topology
+     */
+    public int getTopologyPriority() {
+       return this.topologyPriority;
+    }
+
+    /**
+     * Get the timestamp of when this topology was launched
+     */
+    public int getLaunchTime() {
+        return this.launchTime;
+    }
+
+    /**
+     * Get how long this topology has been executing
+     */
+    public int getUpTime() {
+        return Time.currentTimeSecs() - this.launchTime;
+    }
+
+    @Override
+    public String toString() {
+        return "Name: " + this.getName() + " id: " + this.getId() + " Priority: " + this.getTopologyPriority()
+                + " Uptime: " + this.getUpTime() + " CPU: " + this.getTotalRequestedCpu()
+                + " Memory: " + (this.getTotalRequestedMemOffHeap() + this.getTotalRequestedMemOnHeap());
+    }
+
+    @Override
+    public int hashCode() {
```

Comment:
If we override hashCode we should override equals as well, and have it be consistent with hashCode around the TopologyId.